h2o-py/h2o/automl/_base.py

```diff
@@ -125,3 +126,71 @@ def event_log(self):
         """
         pass
+
+    def get_best_model(self, algorithm=None, criterion=None):
+        """
+        Get best model of a given family/algorithm for a given criterion from an AutoML object.
+
+        :param algorithm: One of "basemodel", "deeplearning", "drf", "gbm", "glm", "stackedensemble", "xgboost".
+                          If None, pick the best model regardless of the algorithm.
+        :param criterion: Criterion can be one of the metrics reported in leaderboard. If set to None, the same ordering
+                          as in the leaderboard will be used.
+                          Avaliable criteria:
+                            - Regression metrics: mean_residual_deviance, rmse, mse, mae, rmsle
+                            - Binomial metrics: auc, logloss, aucpr, mean_per_class_error, rmse, mse
+                            - Multinomial metrics: mean_per_class_error, logloss, rmse, mse, auc, aucpr
+                          The following additional leaderboard information can be also used as a criterion:
+                            - 'training_time_ms': column providing the training time of each model in milliseconds (doesn't include the training of cross validation models).
+                            - 'predict_time_per_row_ms`: column providing the average prediction time by the model for a single row.
+        :return: An H2OModel or None if no model of a given family is present
+        :examples:
+        >>> # Set up an H2OAutoML object
+        >>> aml = H2OAutoML(max_runtime_secs=30)
+        >>> # Launch an AutoML run
+        >>> aml.train(y=y, training_frame=train)
+        >>> gbm = aml.get_best_model("gbm")
+        """
+        from h2o.exceptions import H2OValueError
+        def _get_models(leaderboard):
+            return [m[0] for m in
+                    leaderboard["model_id"].as_data_frame(use_pandas=False, header=False)]
+
+        higher_is_better = ["auc", "aucpr"]
+
+        assert_is_type(algorithm, None, str)
+        assert_is_type(criterion, None, str)
+
+        if criterion is not None:
+            criterion = criterion.lower()
+
+        if algorithm is not None and "any" != algorithm:
+            if algorithm.lower() not in ("basemodel", "deeplearning", "drf", "gbm",
+                                         "glm", "stackedensemble", "xgboost"):
+                raise H2OValueError("Algorithm \"{}\" is not supported!".format(algorithm))
+            algorithm = algorithm.lower()
+            if "basemodel" == algorithm:
+                algorithm = "^(?!stackedensemble)"
+        else:
+            algorithm = ".*"
+
+        extra_cols = ["algo"]
+        if criterion in ("training_time_ms", "predict_time_per_row_ms"):
+            extra_cols.append(criterion)
+
+        leaderboard = h2o.automl.get_leaderboard(self, extra_columns=extra_cols)
+        leaderboard = leaderboard[leaderboard["algo"].tolower().grep(algorithm, output_logical=True), :]
```

Comment:
do you still need grep+regex here?
Can't we just use something like
```
leaderboard = leaderboard[leaderboard["algo"] == algo] if algo else leaderboard
```
I love regexp, but we should avoid them if there is a much simpler alternative.

Reply:
I don't really need it. Thanks!